packages/runtime-dom/src/directives/vOn.ts

```diff
@@ -14,8 +13,17 @@ const modifierGuards: Record<
   left: e => 'button' in e && (e as any).button !== 0,
   middle: e => 'button' in e && (e as any).button !== 1,
   right: e => 'button' in e && (e as any).button !== 2,
-  exact: (e, modifiers) =>
-    modifiers!.some(m => systemModifiers.has(m) && (e as any)[`${m}Key`])
+  exact: (e, modifiers: string[]) => {
+    // todo: replace with makeMap
+    let map = modifiers.reduce(
+      (map, m) => ((map[m] = true), map),
+      Object.create(null)
+    )
+    // Some of the system modifiers are not specified, but is flagged true on the event
```

Comment:
Turns out we have to loop all of the system modifiers like in 2.x (https://github.com/vuejs/vue/blob/dev/src/compiler/codegen/events.js#L131-L138)... If we don't, we can't check for "shiftKey" given modifiers ["a"] or just empty array.
This makeMap is only called when event is triggered, so no render perf impact. It's not slow in run time either.